tools/diagnose.py

```diff
@@ -81,76 +81,98 @@ def test_connection(name, url, timeout=10):
     load_elapsed = time.time() - start
     print("Timing for {}: {}, DNS: {:.4f} sec, LOAD: {:.4f} sec.".format(name, url, dns_elapsed, load_elapsed))
 
+def check_python():
+    print('----------Python Info----------')
+    print('Version      :', platform.python_version())
+    print('Compiler     :', platform.python_compiler())
+    print('Build        :', platform.python_build())
+    print('Arch         :', platform.architecture())
+
+def check_pip():
+    print('------------Pip Info-----------')
+    try:
+        import pip
+        print('Version      :', pip.__version__)
+        print('Directory    :', os.path.dirname(pip.__file__))
+    except ImportError:
+        print('No corresponding pip install for current python.')
+
+def check_mxnet():
+    print('----------MXNet Info-----------')
+    try:
+        import mxnet
+        print('Version      :', mxnet.__version__)
+        mx_dir = os.path.dirname(mxnet.__file__)
+        print('Directory    :', mx_dir)
+        commit_hash = os.path.join(mx_dir, 'COMMIT_HASH')
+        with open(commit_hash, 'r') as f:
+            ch = f.read().strip()
+            print('Commit Hash   :', ch)
+    except ImportError:
+        print('No MXNet installed.')
+    except IOError:
+        pass
+    except:
+        print("An error occured trying to import mxnet.")
+        print("This is very likely due to a missing dll.")
+        raise
```

Comment:
Is there a way to print the error? Sometimes there is useful information such as missing library file or missing symbol.